src/app/modules/payments/__tests__/stripe.service.spec.ts

```diff
@@ -704,4 +705,79 @@ describe('stripe.service', () => {
       expect(result._unsafeUnwrap()).toBe(expectedAccountId)
     })
   })
+
+  describe('handleStripeEvent', () => {
+    describe('event.type of payout', () => {
+      beforeEach(() => jest.resetAllMocks())
+      it('should return error result when call to stripe.balanceTransactions failed', async () => {
+        const processStripeEventSpy = jest
+          .spyOn(StripeService, 'processStripeEvent')
+          .mockImplementationOnce(() => errAsync(new PaymentNotFoundError()))
+        const balanceTransactionApiSpy = jest
+          .spyOn(stripe.balanceTransactions, 'list')
+          .mockImplementationOnce(
+            () =>
+              ({
+                autoPagingEach: () => Promise.reject('boom'),
+              } as unknown as Stripe.ApiListPromise<Stripe.BalanceTransaction>),
+          )
+
+        const result = await StripeService.handleStripeEvent(
+          MOCK_STRIPE_EVENTS_MAP['evt_PAYOUT_CREATED'],
+        )
+        expect(balanceTransactionApiSpy).toHaveBeenCalledOnce()
+        expect(processStripeEventSpy).not.toHaveBeenCalled()
+        expect(result.isErr()).toBeTrue()
+      })
+
+      it('should return error result when processStripeEvent failed with event.type', async () => {
+        const processStripeEventSpy = jest
+          .spyOn(StripeService, 'processStripeEvent')
+          .mockImplementationOnce(() => errAsync(new PaymentNotFoundError()))
+
+        const balanceTransactionApiSpy = jest
+          .spyOn(stripe.balanceTransactions, 'list')
+          .mockImplementationOnce(
+            () =>
+              ({
+                autoPagingEach: (fn) => fn({ type: 'charge', source: {} }),
+              } as unknown as Stripe.ApiListPromise<Stripe.BalanceTransaction>),
+          )
+
+        const result = await StripeService.handleStripeEvent(
+          MOCK_STRIPE_EVENTS_MAP['evt_PAYOUT_CREATED'],
+        )
+
+        expect(balanceTransactionApiSpy).toHaveBeenCalledOnce()
+        expect(processStripeEventSpy).not.toHaveBeenCalledOnce()
```

Comment:
I'm a bit confused by this test - if the test is to check that error result is returned when 'processStripeEvent' fails, doesn't that mean that `processStripeEventSpy` should be called?

Reply:
Great catch. It's should have been called.

Reply:
Added in b919420